docs/best-practices/best-practices.md

```diff
@@ -0,0 +1,370 @@
+# Best practices
+
+Although Earthly has been designed to be unambiguous about what command to use for the job, writing Earthfiles can sometimes still be tricky, when it comes to nuances. As you try to accomplish certain tasks, you may find that sometimes the same result can be achieved using more than one technique. Or so it might seem.
+
+Below we list some of the best practices that we have found to be useful in writing Earthfiles, with a focus on certain commands or techniques that seem similar, but aren't really, but also on some key points that we have seen newcomers stumble into.
+
+## `COPY` only the minimal amount of files. Avoid copying `.git`
+
+A typical mistake is to `COPY` entire large directories into the build environment and only using a subset of the files within them. Or worse, copying the entire repository (which might also include `.git`) for no good reason.
```

Comment:
With Docker you might have a `.dockerignore` to exclude something like `.git`. Do we support something similar?